tt_metal/programming_examples/triple_thread_eltwise_binary/triple_thread_eltwise_binary.cpp

```diff
@@ -0,0 +1,194 @@
+        constexpr uint32_t block_size = 8;
+        constexpr uint32_t dram_buffer_size =
+            single_tile_size * num_tiles;  // num_tiles of FP16_B, hard-coded in the reader/writer kernels
+
+        tt_metal::InterleavedBufferConfig dram_config{
+            .device = device,
+            .size = dram_buffer_size,
+            .page_size = dram_buffer_size,
+            .buffer_type = tt_metal::BufferType::DRAM};
+
+        std::shared_ptr<tt::tt_metal::Buffer> src0_dram_buffer = CreateBuffer(dram_config);
+        std::shared_ptr<tt::tt_metal::Buffer> src1_dram_buffer = CreateBuffer(dram_config);
+        std::shared_ptr<tt::tt_metal::Buffer> dst_dram_buffer = CreateBuffer(dram_config);
+        // Since all interleaved buffers have size == page_size, they are entirely contained in the first DRAM bank
+        uint32_t src0_bank_id = 0;
+        uint32_t src1_bank_id = 0;
+        uint32_t dst_bank_id = 0;
+        /*
+         * Use circular buffers to set input and output buffers that the
+         * compute engine will use.
+         */
+        constexpr uint32_t src0_cb_index = tt::CBIndex::c_0;
+        constexpr uint32_t num_input_tiles = block_size * 2;
+        CircularBufferConfig cb_src0_config =
+            CircularBufferConfig(num_input_tiles * single_tile_size, {{src0_cb_index, tt::DataFormat::Float16_b}})
+                .set_page_size(src0_cb_index, single_tile_size);
+        tt_metal::CreateCircularBuffer(program, core, cb_src0_config);
+
+        constexpr uint32_t src1_cb_index = tt::CBIndex::c_1;
+        CircularBufferConfig cb_src1_config =
+            CircularBufferConfig(num_input_tiles * single_tile_size, {{src1_cb_index, tt::DataFormat::Float16_b}})
+                .set_page_size(src1_cb_index, single_tile_size);
+        tt_metal::CreateCircularBuffer(program, core, cb_src1_config);
+
+        constexpr uint32_t output_cb_index = tt::CBIndex::c_16;
+        constexpr uint32_t num_output_tiles = block_size * 2;
+        CircularBufferConfig cb_output_config =
+            CircularBufferConfig(num_output_tiles * single_tile_size, {{output_cb_index, tt::DataFormat::Float16_b}})
+                .set_page_size(output_cb_index, single_tile_size);
+        tt_metal::CreateCircularBuffer(program, core, cb_output_config);
+
+        /*
+         * Specify data movement kernels for reading/writing data to/from
+         * DRAM.
+         */
+        KernelHandle binary_reader_kernel_id = CreateKernel(
+            program,
+            "tt_metal/kernels/dataflow/reader_binary.cpp",
+            core,
+            DataMovementConfig{.processor = DataMovementProcessor::RISCV_1, .noc = NOC::RISCV_1_default});
+
+        KernelHandle unary_writer_kernel_id = CreateKernel(
+            program,
+            "tt_metal/kernels/dataflow/writer_unary.cpp",
+            core,
+            DataMovementConfig{.processor = DataMovementProcessor::RISCV_0, .noc = NOC::RISCV_0_default});
```

Comment:
DITTO - please put kernels used in examples under the example's folder and annotate them.